tests/unit/maskedOuter.cpp

```diff
@@ -0,0 +1,226 @@
+
+/*
+ *   Copyright 2021 Huawei Technologies Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <utility>
+#include <iostream>
+
+#include "graphblas.hpp"
+
+
+using namespace grb;
+
+// sample data
+static const double vec1_vals[ 3 ] = { 1, 2, 3 };
+static const double vec2_vals[ 3 ] = { 4, 5, 6 };
+
+static const double m1[ 3 ] = { 0.5, 3.4, 5 };
+
+static const size_t I1[ 3 ] = { 0, 1, 2 };
+static const size_t J1[ 3 ] = { 0, 1, 2 };
+
+static const double m2[ 6 ] = { 0.5, -1, 23, 34, -4.4, 3.14 };
+
+static const size_t I2[ 6 ] = { 0, 0, 1, 1, 2, 2 };
+static const size_t J2[ 6 ] = { 1, 2, 0, 2, 0, 1 };
+
+static const double mask_test1_in[ 3 ] = { 1, 1, 1 };
+static const double mask_test1_expect[ 3 ] = { 4, 10, 18 };
+
+static const double mask_test2_in[ 3 ] = { 1, 1, 1 };
+static const double mask_test2_expect[ 3 ] = { 11, 20, 27 };
+
+void grbProgram( const void *, const size_t in_size, int &error ) {
+	error = 0;
+
+	if( in_size != 0 ) {
+		(void)fprintf( stderr, "Unit tests called with unexpected input\n" );
+		error = 1;
+		return;
+	}
+
+	// allocate
+	grb::Vector< double > u( 3 );
+	grb::Vector< double > v( 3 );
+	grb::Matrix< double > Result1( 3, 3 );
+	grb::Matrix< double > Result2( 3, 3 );
+	grb::Matrix< double > mask1( 3, 3 );
+	grb::Matrix< double > mask2( 3, 3 );
+	grb::Vector< double > test1( 3 );
+	grb::Vector< double > out1( 3 );
+	grb::Vector< double > test2( 3 );
+	grb::Vector< double > out2( 3 );
+
+	// semiring
+	grb::Semiring<
+		grb::operators::add< double >, grb::operators::mul< double >,
+		grb::identities::zero, grb::identities::one
+	> ring;
+
+	// initialise vec
+	const double * vec_iter = &(vec1_vals[ 0 ]);
+	grb::RC rc = grb::buildVector( u, vec_iter, vec_iter + 3, SEQUENTIAL );
+	if( rc != SUCCESS ) {
+		std::cerr << "\t initial buildVector FAILED\n";
+		error = 5;
+	}
+
```

Comment:
We recently added the `std::initializer_list<>` constructor for `grb::Vector` if you want 😄 

```suggestion
	grb::Vector< double > u = { 1, 2, 3 };
```

Reply:
Done, thanks!